### PR TITLE
net/lwip: Replace strcpy to strncpy for safety

### PR DESCRIPTION
**Pull request by SeonghoByeon**
- Modify strcpy to strncpy for thread name copy

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [96794ba9606680083e4f0aeb7f837f771fc94027] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [96794ba9606680083e4f0aeb7f837f771fc94027] - Code Rule Check (C++) OK.

File: os/net/lwip/src/api/sockets.c [hunk 1]
```diff
@@ -464,7 +464,7 @@ int copy_socket(void *arg)
 				sock_info->remote_port = 0;
 			}
 			if (pthread_getname_np(sockets[i].conn->pid, sock_info->pid_name)) {
-				strcpy(sock_info->pid_name, "NONE");
```
Comment (sunghan-chang) [inline os/net/lwip/src/api/sockets.c]: Don't you need to add `+ 1` for NULL?

Comment (SeonghoByeon) [reply os/net/lwip/src/api/sockets.c]: sizeof("NONE") returns 5, including the last NULL. Therefore, it automatically copies the last NULL character, as far as I know. Thanks for the comment.

**Comment by seinfra** [comment]
Target : [2dc6c4ab56ee8511b82f7739ff448533ae43e60d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2dc6c4ab56ee8511b82f7739ff448533ae43e60d] - Code Rule Check OK.

**Pull request merged**